src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -47,17 +47,16 @@ protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<Data
     protected FlowSet<DataFlowAbstraction> gen(Unit u, FlowSet<DataFlowAbstraction> in) {
         FlowSet<DataFlowAbstraction> res = new ArraySparseSet<>();
         if (isSourceStatement(u) || isSinkStatement(u)) {
-            u.getDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
+            u.getDefBoxes().stream().filter(v -> v.getValue() instanceof Local || v.getValue() instanceof JArrayRef).forEach(v -> {
                 Statement stmt = isSourceStatement(u)
                         ? findSourceStatement(u)
                         : findSinkStatement(u);
-                res.add(new DataFlowAbstraction((Local) v.getValue(), stmt));
-            });
-            u.getDefBoxes().stream().filter(v -> v.getValue() instanceof JArrayRef).forEach(v -> {
-                Statement stmt = isSourceStatement(u)
-                        ? findSourceStatement(u)
-                        : findSinkStatement(u);
-                res.add(new DataFlowAbstraction((Local) getBaseBoxName(v), stmt));
+
+                Local local = v.getValue() instanceof Local
+                        ? (Local) v.getValue()
+                        : (Local) getJArrayRefName(v);
```

Comment:
Why the casting is Local and not JArrayRef?

Reply:
JArrayRef is a LOCAL, if you change the casting, you will have to create a new attribute and constructor in DataFlowAbstraction class.

Reply:
JArrayRef is a Local, if you change the casting, you will have to create a new attribute and constructor in DataFlowAbstraction class.

Reply:
> Why the casting is Local and not JArrayRef?


because DataFlowAbstraction receives a place and a statement.

Reply:
> JArrayRef is a LOCAL, if you change the casting, you will have to create a new attribute and constructor in DataFlowAbstraction class.

Yes you are right. I preferred not to make changes to the DataFlowAbstraction and used this cast that worked perfectly.

Reply:
A JArrayRef has also a Local object, y[1] is a JAarayRef object and only the name y is a Local object.

Reply:
`JArrayRef ` is a class that is instantiated by a **base** and an **index**, this is the biggest difference for `Local`.

E.g: `y​​[1]`
y = base
[1] = index

`Local` is an interface and represents local variables as I see it.

The base attribute of `JArrayRef` is a `ValueBox` and can be converted to `Local` via casting. I used these strategies to avoid making changes to `DataFlowAbstraction` and having to add a new specific attribute for `JArrayRef`.

Reply:
@barbosamaatheus what do you mean by "base attribute"?